### ISSUE TITLE
[RN73] Support for iOS

### ISSUE THREAD

**Issue by d4vidi** [wix/react-native-navigation#7836]

**Comment by SudoPlz** [comment]
hey @d4vidi can I help with that?
I could open a PR if you're willing to review, because I'm currently blocked on this.

**Comment by gosha212** [comment]
Hi @SudoPlz,
I'm currently working on this task. It will be ready soon. This task is hard to estimate because it is a research task.

**Comment by gosha212** [comment]
@SudoPlz I did the Andorid part and currently working on the typescript. Hope to get to it very soon. I will share a link to a draft PR in the beginning of the next week.

**Comment by oblador** [comment]
@SudoPlz you can get past that by replacing the fabric if statement in the podspec with:
```ruby
  if fabric_enabled
    install_modules_dependencies(s)
    s.requires_arc    = true
  end
```

**Comment by gosha212** [comment]
@SudoPlz https://github.com/wix/react-native-navigation/pull/7844 here is the PR

**Comment by oblador** [comment]
@gosha212 Awesome – thank you, however it only fixes Android? iOS Fabric is still broken for me.

**Comment by gosha212** [comment]
@oblador I didn't test it with Fabric yet. The tests are working on both platforms

**Comment by oblador** [comment]
Ah, ok. @SudoPlz's comment is about Fabric on iOS hence the confusion.

**Comment by gosha212** [comment]
@oblador The code works with Fabric on Android. IOS still WIP

**Comment by gosha212** [comment]
@oblador the latest commit should work with Fabric. The fix is not merged yet.

**Comment by asafkorem** [comment]
> @gosha212 Thanks! I tried it out and it does not work with fabric and RN 0.73.4 for me. In the pod spec you are overwriting all the config set by react native. I got everything working with this more minimal setup:
> 
> ```
>   if fabric_enabled
>     s.pod_target_xcconfig = {
>       'HEADER_SEARCH_PATHS' => '"$(PODS_ROOT)/Headers/Private/React-Core"',
>     }
>     s.requires_arc = true
>     install_modules_dependencies(s)
>   end
> ```

Hey @oblador, in the podspec we just add React's headers to the search paths. What error are you getting? can you please copy it here?

**Comment by asafkorem** [comment]
@SudoPlz @oblador please try the latest released version ([7.38.0](https://github.com/wix/react-native-navigation/releases/tag/7.38.0))

**Comment by oblador** [comment]
@asafkorem That version fails for me during install with:
```
node_modules/react-native-navigation: Command failed.
Exit code: 1
Command: node scripts/postinstall.js
```

The issue is overwriting search paths and c++ flags causing my build to fail for failing to import `memory` et al. `install_modules_dependencies` will append, therefore it should be placed after not before your own configs

**Comment by oblador** [comment]
@asafkorem have a fix for the postinstall bug here: https://github.com/wix/react-native-navigation/pull/7849

**Comment by asafkorem** [comment]
Thanks @oblador, checking 🙏🏼

**Comment by SudoPlz** [comment]
Waiting for the merged code to be released. 0.38 fails on my end too because of the post install script.

**Comment by asafkorem** [comment]
@oblador @SudoPlz please try [7.38.1](https://github.com/wix/react-native-navigation/releases/tag/7.38.1)

**Comment by SudoPlz** [comment]
I did, everything builds fine. The only problem is, I get a white screen and no react components show up.
Still troubleshooting.

**Comment by oblador** [comment]
Yup same for me, it installs and builds now, but I'm seeing an empty screen.

**Comment by SudoPlz** [comment]
so glad it's not just me.

**Comment by asafkorem** [comment]
@oblador @SudoPlz 
We'll look into this. Does the issue occur on iOS, Android, or both? Also, does this issue persist in the most recent version when Fabric is disabled as well, or only when Fabric is enabled?

**Comment by SudoPlz** [comment]
> @oblador @SudoPlz We'll look into this. Does the issue occur on iOS, Android, or both? Also, does this issue persist in the most recent version when Fabric is disabled as well, or only when Fabric is enabled?

I've only tested iOS so far.

It **only** happens when fabric is enabled. (when it's disabled everything works fine)

**Comment by asafkorem** [comment]
OK, thanks for reporting. I'll test that with a clean RN+RNN app.
Also, can you please list the other public dependencies you're using in your app? I wonder if there might be another suspect. Our playground app plays well with the latest version 🤔

**Comment by asafkorem** [comment]
FYI @gosha212 @d4vidi @SudoPlz @oblador 

I'm able to reproduce this issue on iOS with a [new RN 0.73.4 + Fabric + RNN 7.38.1 app
](https://github.com/asafkorem/react-native-navigation-demo)When disabling fabric - app works fine. For some reason it doesn't reproduce on [our playground app](https://github.com/wix/react-native-navigation/tree/master/playground).

I'll continue the investigation later today. Will update.

**Comment by SudoPlz** [comment]
@asafkorem super happy you were able to repro this one.
Did you track the causing issue by any chance?

**Comment by SudoPlz** [comment]
@asafkorem any news on that?

**Comment by asafkorem** [comment]
Hi @SudoPlz, I made a brief investigation to identify the underlying issue but haven't uncovered anything so far :(
At the moment, we aren't focused on this, but we plan to revisit it shortly.

**Comment by SudoPlz** [comment]
Hey @asafkorem, no pressure or anything, but I'm curious, any rough estimates on when you'll get to work on that issue?

**Comment by SudoPlz** [comment]
@asafkorem or is there any way I can help contribute to help move along the migration?
It's important for me to start working towards fabric, and I'm not confident I understand what's blocking it at the moment, so if there's anything I can do to help I'd be happy to.

**Comment by asafkorem** [comment]
Hey @SudoPlz, [this issue](https://github.com/wix/react-native-navigation/issues/7836#issuecomment-1953600080) requires investigation and we're not working on it at the moment. We plan to get there soon but for the meanwhile feel free to clone the reproduction repo https://github.com/asafkorem/react-native-navigation-demo and investigate why we get this white screen when Fabric is enabled. 
Thanks for the willingness to help! 

Also, if we'll have any update from our end we'll post it here.

**Comment by SudoPlz** [comment]
Hey @d4vidi why was this closed since it's still buggy?
Is there another issue responsible for fixing React Native fabric?

**Comment by d4vidi** [comment]
> Hey @d4vidi why was this closed since it's still bugging? Is there another issue responsible for fixing React Native fabric?

Hey @SudoPlz, yes - we will deal with fabric soon on a different context, in due time. Contribution (upgrading and surfacing the issues) would definitely help speed things up 🙏🏻

**Comment by gosha212** [comment]
@SudoPlz here is a ticket for tracking: #7836
As @d4vidi mentioned. We will be happy to get some contribution here =)

**Comment by TheLonelyAstronaut** [comment]
@d4vidi @gosha212 added fix in ticket [#7836](https://github.com/wix/react-native-navigation/issues/7836)
